src/hera/workflows/script.py

```diff
@@ -232,16 +236,41 @@ def _build_script(self) -> _ModelScriptTemplate:
 def _get_parameters_from_callable(source: Callable) -> Optional[List[Parameter]]:
     # If there are any kwargs arguments associated with the function signature,
     # we store these as we can set them as default values for argo arguments
-    source_signature: Dict[str, Optional[object]] = {}
+    #
+    # Take all the annotation information and make sure it is stored.
```

Comment:
Comment not relevant here - something like this might be helpful above the `get_args` below
```py
        # Take parameter attribute values from annotation
        annotation = get_args(p.annotation)[1]
        for attr in ["enum", "description", "default", "name"]:
```

(But the code is quite self explanatory so it's up to you - comment would be helpful to newbies)